cpp/bench/ann/src/common/util.hpp

```diff
@@ -90,9 +90,15 @@ struct buf {
 
 struct cuda_timer {
  private:
+#ifndef CPU_ONLY
   cudaStream_t stream_{nullptr};
   cudaEvent_t start_{nullptr};
   cudaEvent_t stop_{nullptr};
+#else
+  cudaStream_t stream_{0};
+  cudaEvent_t start_{0};
+  cudaEvent_t stop_{0};
+#endif
```

Comment:
As I noted above, I think it's better to typedef these two as `void*`, so there's no need for extra pragmas here. Alternatively, you can just change the curly braces to the parentheses to avoid the type conversion error popping up.